src/test/kotlin/racingcar/service/RacingCarGameMachineTest.kt

```diff
@@ -12,42 +15,54 @@ class RacingCarGameMachineTest : ShouldSpec({
     context("게임을 진행한다.") {
         should("일반적인 게임을 진행한다.") {
             val sut = RacingCarGameMachine(RandomNumberZeroToNineGenerate())
```

Comment:
`NumberGenerateStrategy`를 fun interface로 정의하셨으니, 해당 테스트 코드는 아래와 같이 람다 표현식으로 변경해볼 수도 있을 것 같아요. 😃
```suggestion
            val sut = RacingCarGameMachine { 0 }
```

Reply:
중첩하는 클래스를 한번 작성해보고싶었어요 :) 감사합니다